src/t8_netcdf.c

```diff
@@ -0,0 +1,87 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <t8.h>
+#include <t8_netcdf.h>
+
+/* Create an extern NetCDF variable */
+t8_netcdf_variable_t *
+t8_netcdf_create_var (t8_netcdf_variable_type_t var_type,
+                      const char *var_name, const char *var_long_name,
+                      const char *var_unit, sc_array_t * var_data)
+{
+#if T8_WITH_NETCDF
+  T8_ASSERT (0 <= var_type && var_type <= 2);
+  t8_netcdf_variable_t *netcdf_variable = T8_ALLOC (t8_netcdf_variable_t, 1);
+  netcdf_variable->datatype = var_type;
+  netcdf_variable->variable_name = var_name;
+  netcdf_variable->variable_long_name = var_long_name;
+  netcdf_variable->variable_units = var_unit;
+  netcdf_variable->var_user_data = var_data;
+  return netcdf_variable;
+#else
+  t8_global_errorf
+    ("This version of t8code is not compiled with netCDF support.\n");
+  return NULL;
+#endif
+}
+
+/* Create an extern NetCDF integer variable */
+t8_netcdf_variable_t *
+t8_netcdf_create_integer_var (const char *var_name, const char *var_long_name,
+                              const char *var_unit, sc_array_t * var_data)
+{
+#if T8_WITH_NETCDF
+  t8_netcdf_variable_type_t var_type;
+  /* Check wheter 32-bit (4-byte) integer or 64-bit integer data sholud be written */
+  var_type = (var_data->elem_size > 4) ? T8_NETCDF_INT64 : T8_NETCDF_INT;
```

Comment:
Are those the only 2 cases that can happen?

Reply:
Currently, yes. These are the only cases which we have implemented so far. But this evaluation has to be adapted if for example an option for `NC_SHORT` variables (signed 2-byte integer) is integrated.